tests/pulse/test_parametrized_evolution.py

```diff
@@ -36,8 +36,48 @@ class MyOp(qml.RX):  # pylint: disable=too-few-public-methods
     has_diagonalizing_gates = False
 
 
+def amp(p, t):
+    return p * t
+
+
+H = qml.PauliX(1) + amp * qml.PauliZ(0) + amp * qml.PauliY(1)
```

Comment:
Could we also add a Hamiltonian with a ragged param shape? Something like one coefficient has 1 parameter and the other has 2.